contrib/slurm/slurmexecutor.py

```diff
@@ -0,0 +1,324 @@
+        It also calls functions for output before and after the run.
+        """
+        self.output_handler.output_before_run(run)
+
+        args = run.cmdline()
+        logging.debug("Command line of run is %s", args)
+
+        try:
+            with open(run.log_file, "w") as f:
+                for i in range(6):
+                    f.write(os.linesep)
+
+            run_result = run_slurm(
+                self.benchmark,
+                args,
+                run.log_file,
+            )
+
+        except KeyboardInterrupt:
+            # If the run was interrupted, we ignore the result and cleanup.
+            stop()
+
+        if STOPPED_BY_INTERRUPT:
+            try:
+                if self.benchmark.config.debug:
+                    os.rename(run.log_file, run.log_file + ".killed")
+                else:
+                    os.remove(run.log_file)
+            except OSError:
+                pass
+            return 1
+
+        run.set_result(run_result)
+        self.output_handler.output_after_run(run)
+        return None
+
+
+jobid_pattern = re.compile(r"job (\d*) queued")
+
+
+def run_slurm(benchmark, args, log_file):
+    timelimit = benchmark.rlimits.cputime
+    cpus = benchmark.rlimits.cpu_cores
+    memory = benchmark.rlimits.memory
+
+    srun_timelimit_h = int(timelimit / 3600)
+    srun_timelimit_m = int((timelimit % 3600) / 60)
+    srun_timelimit_s = int(timelimit % 60)
+    srun_timelimit = f"{srun_timelimit_h}:{srun_timelimit_m}:{srun_timelimit_s}"
+
+    mem_per_cpu = int(memory / cpus / 1000000)
+
+    assert benchmark.config.scratchdir and os.path.exists(benchmark.config.scratchdir)
+
+    with tempfile.TemporaryDirectory(dir=benchmark.config.scratchdir) as tempdir:
+
+        os.makedirs(os.path.join(tempdir, "upper"))
+        os.makedirs(os.path.join(tempdir, "work"))
+
+        singularity_command = (
+            (
+                [
+                    "singularity",
+                    "exec",
+                    "-B",
+                    "./:/lower",
+                    "--no-home",
+                    "-B",
+                    f"{tempdir}:/overlay",
+                    "--fusemount",
+                    f"container:fuse-overlayfs -o lowerdir=/lower -o upperdir=/overlay/upper -o workdir=/overlay/work /home/{os.getlogin()}",
+                    benchmark.config.singularity,
+                ]
+                + args
+            )
+            if benchmark.config.singularity
+            else args
+        )
+        srun_command = [
+            "srun",
+            "-t",
+            str(srun_timelimit),
+            "-c",
+            str(cpus),
+            "-o",
+            str(log_file),
+            "--mem-per-cpu",
+            str(mem_per_cpu),
+            "--threads-per-core=1",  # --use_hyperthreading=False is always given here
+            "--ntasks=1",
+        ] + singularity_command
+        logging.debug("Command to run: %s", " ".join(srun_command))
```

Comment:
If we use `" ".join(map(util.escape_string_shell, srun_command))` here and import `benchexec.util`, you should be able to copy-and-paste the command from the log and execute it in your terminal, even in case of spaces and other special characters inside parameters.

Reply:
Good to know this exists! Will do.